plugins/modules/aci_access_span_src_group_src.py

```diff
@@ -0,0 +1,442 @@
+            child_configs.append({"spanRsSrcToEpg": {"attributes": {"tDn": epg_dn}}})
+        elif routed_outside:
+            # encap is set to unknown when not provided to ensure change is executed and detected properly on update
+            encap = "vlan-{0}".format(routed_outside.get("encap")) if routed_outside.get("encap") else "unknown"
+            if routed_outside.get("tenant") and routed_outside.get("l3out"):
+                l3ext_out_dn = "uni/tn-{0}/out-{1}".format(routed_outside.get("tenant"), routed_outside.get("l3out"))
+            else:
+                # tDn is set to "" when not provided to ensure change is executed and detected properly on update
+                l3ext_out_dn = ""
+            child_configs.append({"spanRsSrcToL3extOut": {"attributes": {"encap": encap, "tDn": l3ext_out_dn}}})
+
+        # Validate if existing and remove child objects when do not match provided configuration
+        if isinstance(aci.existing, list) and len(aci.existing) > 0:
+            # Commented validate code to avoid making additional API request which is handled by APIC error
+            # Keeping for informational purposes
+            # Validate drop_packets are set on parent correctly
+            # if aci.get_config_request("{0}/rssrcGrpToFilterGrp.json".format(source_group_path)) != [] and drop_packets:
+            #     module.fail_json(msg="It is not allowed to configure 'drop_packets: true' when a filter group is configured on the source group.")
+
+            source_path = "/api/mo/uni/infra/srcgrp-{0}/src-{1}".format(source_group, source)
+            for child in aci.existing[0].get("spanSrc", {}).get("children", {}):
+                if child.get("spanRsSrcToFilterGrp") and child.get("spanRsSrcToFilterGrp").get("attributes").get("tDn") != filter_group_tdn:
+                    # Appending to child_config list not possible because of APIC Error 103: child (Rn) of class spanRsSrcGrpToFilterGrp is already attached.
+                    # A seperate delete request to dn of the spanRsSrcGrpToFilterGrp is needed to remove the object prior to adding to child_configs.
+                    # Failed child_config is displayed in below:
+                    #
+                    # child_configs.append(
+                    #     {
+                    #         "spanRsSrcGrpToFilterGrp": {
+                    #             "attributes": {
+                    #                 "dn": "uni/infra/srcgrp-{0}/src-{1}/rssrcGrpToFilterGrp".format(source_group, source),
+                    #                 "status": "deleted",
+                    #             }
+                    #         }
+                    #     }
+                    # )
+                    aci.delete_config_request("{0}/rssrcToFilterGrp.json".format(source_path))
+                elif child.get("spanRsSrcToEpg") and child.get("spanRsSrcToEpg").get("attributes").get("tDn") != epg_dn:
```

Comment:
isn't there a case when both spanRsSrcToFilterGrp and spanRsSrcToEpg or spanRsSrcToL3extOut are changed?
Shouldn't the elif be if's?

Reply:
Not sure what you exactly mean by your comment but elif is there because the child object is only of one class type. So all of the scenarios are caught in the loop, where it checks each child object in the aci.existing response and deletes when the target dn is not matching.

Reply:
I see. The loop process one child at a time so even if i have two child changes, they each run through the if/else on separate loop iteration.